crates/torii/indexer/src/processors/erc1155_transfer_batch.rs

```diff
@@ -0,0 +1,110 @@
+use anyhow::Error;
+use async_trait::async_trait;
+use cainome::cairo_serde::{CairoSerde, U256 as U256Cainome};
+use dojo_world::contracts::world::WorldContractReader;
+use starknet::core::types::{Event, U256};
+use starknet::providers::Provider;
+use torii_sqlite::Sql;
+use tracing::debug;
+
+use super::{EventProcessor, EventProcessorConfig};
+use crate::task_manager::{self, TaskId, TaskPriority};
+
+pub(crate) const LOG_TARGET: &str = "torii_indexer::processors::erc1155_transfer_batch";
+
+#[derive(Default, Debug)]
+pub struct Erc1155TransferBatchProcessor;
+
+#[async_trait]
+impl<P> EventProcessor<P> for Erc1155TransferBatchProcessor
+where
+    P: Provider + Send + Sync + std::fmt::Debug,
+{
+    fn event_key(&self) -> String {
+        "TransferBatch".to_string()
+    }
+
+    fn validate(&self, event: &Event) -> bool {
+        // key: [hash(TransferBatch), operator, from, to]
+        // data: [ids_len, ids[0].low, ids[0].high, ..., values_len, values[0].low, values[0].high,
+        // ...]
+        if event.keys.len() == 4 && !event.data.is_empty() {
+            return true;
+        }
+        false
+    }
+
+    fn task_priority(&self) -> TaskPriority {
+        1
+    }
+
+    fn task_identifier(&self, _event: &Event) -> TaskId {
+        task_manager::TASK_ID_SEQUENTIAL
+    }
+
+    async fn process(
+        &self,
+        _world: &WorldContractReader<P>,
+        db: &mut Sql,
+        block_number: u64,
+        block_timestamp: u64,
+        event_id: &str,
+        event: &Event,
+        _config: &EventProcessorConfig,
+    ) -> Result<(), Error> {
+        let token_address = event.from_address;
+        let from = event.keys[2];
+        let to = event.keys[3];
+
+        let ids_len = event.data[0].try_into().unwrap_or(0u64) as usize;
+        let mut current_idx = 1;
+
+        // First pass: read all token IDs
```

Comment:
May add some comment on what's the expected buffer content for multiple tokens transfer here or link to the ERC spec for quicker understanding.

Reply:
it is specified on the validate; `// data: [ids_len, ids[0].low, ids[0].high, ..., values_len, values[0].low, values[0].high` but i can add a reference there too